### PR TITLE
OCPBUGS-5063: E2E: Network stack Pinning tests

### PR DESCRIPTION
**Pull request by mrniranjan**
1. Tests to check rps mask when reserved cpus list is modified
2. Tests to check if rps mask is not applied when realtimeHint is disabled

Signed-off-by: Niranjan M.R <mrniranjan@redhat.com>

### CODE REVIEW

**Comment by MarSik** [comment]
Looks good to me at the first glance. I need to think a bit about possible race conditions wrt the veth queues. Even though the values should be pretty stable as we are not starting any pods.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 1]
```diff
@@ -1348,6 +1348,163 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
...
+					}
+				}
+				rpsMaskContent := strings.TrimSuffix(maskContent, "\r")
+				Expect(len(strings.Split(rpsMaskContent, " "))).To(Equal(3), "systemd unit file doesn't have proper rpsmask")
+				serviceRPSCPUs := strings.Split(rpsMaskContent, " ")[2]
+				rpsCPUs, err := components.CPUMaskToCPUSet(serviceRPSCPUs)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(rpsCPUs).To(Equal(expectedRPSCPUs), "the service rps mask is different from the reserved CPUs")
+
+				// Verify all host network devices have the correct RPS mask
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1456]: We already have a similar test for that right? is it different than the old one?

**Review by Tal-or — COMMENTED** [review]
In general looks good to me. 
Just a couple of comments about readability from my side.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go (hunk 1 above)
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go:1456]: Yes its kind of similar where we are testing rps mask. The difference here, we are testing rps mask where we are modifying the reserved cpus and we are using cpus from the middle of the avaiable cpu id's.  Once the reserved cpus are updated we are checking if the rps mask got applied as per our cpus that we picked up for reserved.

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 2]
```diff
@@ -1348,6 +1348,163 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
...
+				RealTime:              pointer.BoolPtr(false),
+				PerPodPowerManagement: pointer.BoolPtr(false),
+			}
+
+			By("Patching the performance profile with workload hints")
+			workloadHints, err := json.Marshal(profile.Spec.WorkloadHints)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Updating the performance profile")
+			Expect(testclient.Client.Patch(context.TODO(), profile,
```
[the remark this answers is not in the document]
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @Tal-or i have addressed in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 3]
```diff
@@ -1348,6 +1348,163 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
...
+
+			By("Backing up the profile")
+			initialProfile = profile.DeepCopy()
+		})
+
+		AfterEach(func() {
+			//Revert the profile
+			spec, err := json.Marshal(initialProfile.Spec)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testclient.Client.Patch(context.TODO(), profile,
```
[the remark this answers is not in the document]
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @Tal-or i have addressed in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 4]
```diff
@@ -1348,6 +1348,163 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
...
+				for k := range numaCoreSiblings[key] {
+					cores = append(cores, k)
+				}
+				sort.Ints(cores)
+				middleCoreIds := cores[len(cores)/2]
+				siblings := nodes.GetCpuSiblings(numaCoreSiblings, middleCoreIds)
+				reserved = append(reserved, siblings...)
+			}
+			reservedCpus := strings.Join(reserved, ",")
+			for key := range numaCoreSiblings {
```
[the remark this answers is not in the document]
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @Tal-or i have addressed in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 5]
```diff
@@ -1348,6 +1348,163 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
...
+				numaCoreSiblings, err = nodes.GetCoreSiblings(&node)
+			}
+			//Lets take reserved cpus from the middle of the cpu list
+			for key := range numaCoreSiblings {
+				cores := make([]int, 0)
+				for k := range numaCoreSiblings[key] {
+					cores = append(cores, k)
+				}
+				sort.Ints(cores)
+				middleCoreIds := cores[len(cores)/2]
```
[the remark this answers is not in the document]
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @Tal-or i have addressed in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 6]
```diff
@@ -1348,6 +1348,163 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
...
+			}
+			//numaNode[node][coreId][core-siblings]
+			var numaCoreSiblings map[int]map[int][]int
+			for _, node := range workerRTNodes {
+				numaCoreSiblings, err = nodes.GetCoreSiblings(&node)
+			}
+			//Lets take reserved cpus from the middle of the cpu list
+			for key := range numaCoreSiblings {
+				cores := make([]int, 0)
+				for k := range numaCoreSiblings[key] {
```
[the remark this answers is not in the document]
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @Tal-or i have addressed in the latest patch

File: test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go [hunk 7]
```diff
@@ -1348,6 +1348,163 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
...
+					Skip(fmt.Sprintf("This test needs more than 8 CPUs online to work correctly, current online CPUs are %s", onlineCPUCount))
+				}
+			}
+			//numaNode[node][coreId][core-siblings]
+			var numaCoreSiblings map[int]map[int][]int
+			for _, node := range workerRTNodes {
+				numaCoreSiblings, err = nodes.GetCoreSiblings(&node)
+			}
+			//Lets take reserved cpus from the middle of the cpu list
+			for key := range numaCoreSiblings {
```
[the remark this answers is not in the document]
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go]: @Tal-or i have addressed in the latest patch

**Comment by Tal-or** [comment]
@mrniranjan Thank you for your changes.
/lgtm

**Comment by Tal-or** [comment]
/lgtm
@mrniranjan Thank you for your changes.

**Comment by fromanirh** [comment]
/approve
we totally want this automation

**Review by ffromani — APPROVED** [review]

**Pull request merged**